### PR TITLE
Speed up mpsc_stress test

### PR DESCRIPTION
**Pull request by mejrs**
See https://rust-lang.zulipchat.com/#narrow/stream/122651-general/topic/mpsc_stress for context

r? windows

### CODE REVIEW

**Comment by compiler-errors** [comment]
r? compiler

I'm guessing people in the windows ping group don't have bors rights

**Review by ChrisDenton — COMMENTED** [review]
I just have a couple of comments. But this looks fine.

File: src/test/ui/threads-sendsync/mpsc_stress.rs [hunk 1]
```diff
@@ -159,7 +155,62 @@ fn concurrent_writes_iter() {
...
+fn with_minimum_timer_resolution(f: impl Fn()) {
+    #[cfg(windows)]
+    unsafe {
+        let mut caps = timeapi::timecaps_tag {
+            wPeriodMin: 0,
+            wPeriodMax: 0,
+        };
+
+        let ret =
+            timeapi::timeGetDevCaps(&mut caps, std::mem::size_of::<timeapi::timecaps_tag>() as _);
```
[the remark this answers is not in the document]
Comment (thomcc) [reply src/test/ui/threads-sendsync/mpsc_stress.rs:200]: This definitely feels like overkill, and forces us to include a bunch of extra manual FFI bindings which complicates maintenance. `timeBeginPeriod(1)` should always work, so I'd be inclined to use that.

File: src/test/ui/threads-sendsync/mpsc_stress.rs [hunk 2]
```diff
@@ -159,7 +155,62 @@ fn concurrent_writes_iter() {
...
+        assert_eq!(ret, timeapi::TIMERR_NOERROR);
+
+        let ret = timeapi::timeBeginPeriod(caps.wPeriodMin);
+        assert_eq!(ret, timeapi::TIMERR_NOERROR);
```
[the remark this answers is not in the document]
Comment (mejrs) [reply src/test/ui/threads-sendsync/mpsc_stress.rs:204]: I'm happy to gut the other things out, but I prefer to err on the side of checking, especially if this should never happen (i've been bitten by that a few times)

[the remark this answers is not in the document]
Comment (ChrisDenton) [reply src/test/ui/threads-sendsync/mpsc_stress.rs:204]: That's fair.

File: src/test/ui/threads-sendsync/mpsc_stress.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mejrs) [reply src/test/ui/threads-sendsync/mpsc_stress.rs]: I've deleted it.

**Review by ChrisDenton — APPROVED** [review]
LGTM, thanks!

**Pull request merged**